src/app/testing/auth-service-stub.ts

```diff
@@ -1,9 +1,20 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
 export class AuthServiceStub {
-  user: string|null = 'test_user@test.com';
+  user: string|undefined = 'test_user@test.com';
 
-  async logInOrOut(): Promise<void> {}
+  async logInOrOut(): Promise<void> {
```

Comment:
Could you make the stubbed auth service inherit logInOrOut from the real service? As is the stubbed authService logInOrOut function looks identical to the real one, but if someone were to make a change to the real function, and not realize they needed to change the stub as well, tests would still pass. Typically a pattern we use is to have a base service and two services that inherit from it - a prod and a mock (or stub) service. Putting this sort of functionality in the base service and then leaving the implementation of the functions that deal with the real async stuff up to each implementation is a paradigm that has worked well for us.

Reply:
I guess that comment had two conflicting suggestions in it - having both services inherit from a base service probably makes sense here.

Reply:
Created an AuthServiceBase and had the stub/production version inherit from it.